lib/Factory/LayoutFactory.php

```diff
@@ -1362,6 +1378,29 @@ public function createFromZip(
         $widgets = $layout->getAllWidgets();
         $this->getLog()->debug('Layout has ' . count($widgets) . ' widgets');
 
+        // Go through all the widgets and upgrade from v3 to v4
+        foreach ($widgets as $widget) {
+            // Load the widget
+            $widget->load();
+
+            $module = $this->moduleFactory->getByType($widget->type);
+            if ($module->isWidgetCompatibilityAvailable()) {
+                // Grab a widget compatibility interface, if there is one
+                $widgetCompatibilityInterface = $module->getWidgetCompatibilityOrNull();
+                if ($widgetCompatibilityInterface !== null) {
+                    try {
+                        $widgetCompatibilityInterface->upgradeWidget($widget, $widget->schemaVersion, 2);
+                        if ($widget->schemaVersion == 1) {
```

Comment:
We should only update the schema version if `upgradeWidget` actually did something (i.e. returns true). That way, if we do later make changes to the widget and decide to update its schema, we can run the task/import again and have it upgraded.